R/teal_slice.R

```diff
@@ -483,50 +512,76 @@ c.teal_slices <- function(...) {
   )
 }
 
+#' @param tss `teal_slices` object
+#' @param file `character(1)` specifying path to save to
+#' @export
+#' @rdname teal_slice
+#' @keywords internal
+#'
+store_slices <- function(tss, file) {
+  checkmate::assert_class(tss, "teal_slices")
+  checkmate::assert_path_for_output(file, overwrite = TRUE, extension = "json")
 
-# format method for teal_slices
+  cat(format(tss, show_all = TRUE, nchar = NULL), "\n", file = file)
+}
+
+#' @param file `character(1)` specifying path to read from
 #' @export
 #' @rdname teal_slice
 #' @keywords internal
 #'
-format.teal_slices <- function(x, show_all = FALSE, ...) {
-  res <- character(0)
-  for (i in seq_along(x)) {
-    ind <- names(x)[i]
-    if (is.null(ind)) ind <- sprintf("[[%d]]", i)
-    res <- append(res, ind)
-    res <- append(res, format(x[[i]], show_all = show_all, ...))
-  }
+restore_slices <- function(file) {
+  checkmate::assert_file_exists(file, access = "r", extension = "json")
 
-  includes <- attr(x, "include_varnames")
-  if (length(includes) > 0L) {
-    res <- append(res, "\nfilterable variables:")
-    for (i in seq_along(includes)) {
-      res <- append(res, sprintf(" $ %s: %s", names(includes)[i], toString(includes[[i]])))
-    }
-  }
+  tss_j <- jsonlite::fromJSON(file, simplifyDataFrame = FALSE)
 
-  excludes <- attr(x, "exclude_varnames")
-  if (length(excludes) > 0L) {
-    res <- append(res, "\nnon-filterable variables:")
-    for (i in seq_along(excludes)) {
-      res <- append(res, sprintf(" $ %s: %s", names(excludes)[i], toString(excludes[[i]])))
-    }
-  }
+  tss_elements <-
+    lapply(tss_j$slices, function(x) {
+      as.teal_slice(Filter(Negate(is.null), x))
+    })
 
-  ct <- attr(x, "count_type")
-  res <- append(res, sprintf("\ncount type: %s", ct))
-  paste(c(res, ""), collapse = "\n")
+  tss <- do.call(filter_settings, c(`...` = tss_elements, tss_j$attributes))
+  names(tss) <- NULL
+  tss
 }
 
+#' @param x `teal_slice` object
```

Comment:
```suggestion
```
Again, do not document arguments to methods.

Reply:
duly noted